### PR TITLE
Fix typescript integration bug with ajv-compiler

### PR DESCRIPTION
**Pull request by Uzlopak**
should resolve https://github.com/fastify/ajv-compiler/issues/95

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: types/schema.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { ValidatorCompiler } from '@fastify/ajv-compiler'
+import { ValidatorFactory } from '@fastify/ajv-compiler'
```
Comment (Uzlopak) [inline types/schema.d.ts:1]: I was not that sure, regarding this typing change.

**Review by mcollina — APPROVED** [review]
Lgtm

File: types/schema.d.ts (hunk 1 above)
Comment (Eomm) [reply types/schema.d.ts:1]: But isn't it the same?

https://github.com/fastify/ajv-compiler/blob/c257dc65ed2b1e9213303fc8caec39ff40566a3b/types/index.d.ts#L54

Comment (Uzlopak) [reply types/schema.d.ts:1]: No. The reason why it currently doesnt work properly is that the ValidatorFactory instantiates  ValidatorCompilers.

The reason why i am unsure is that i would like to have atleast a typing test.  Other than that i think it should be the ValidatorFactory.

**Review by Eomm — APPROVED** [review]

**Pull request merged**